src/app/public/modules/checkbox/checkbox.component.ts

```diff
@@ -2,58 +2,38 @@
 import {
   Component,
   EventEmitter,
-  forwardRef,
   Input,
-  Output
+  OnChanges,
+  OnInit,
+  Optional,
+  Output,
+  Self,
+  SimpleChanges
 } from '@angular/core';
 
 import {
   AbstractControl,
   ControlValueAccessor,
-  NG_VALUE_ACCESSOR,
-  ValidationErrors,
-  Validator,
-  NG_VALIDATORS
+  NgControl,
+  Validators
 } from '@angular/forms';
 
 /**
  * Monotonically increasing integer used to auto-generate unique ids for checkbox components.
  */
 let nextId = 0;
 
-/**
- * Provider Expression that allows sky-checkbox to register as a ControlValueAccessor.
- * This allows it to support [(ngModel)].
- */
-// tslint:disable:no-forward-ref no-use-before-declare
-export const SKY_CHECKBOX_CONTROL_VALUE_ACCESSOR: any = {
```

Comment:
Aren't the accessors and validators required?

Reply:
Not anymore. Now that we're injecting the `NgControl`, it already comes with those providers. All we need to do is set the value accessor to the checkbox component. See this line in my constructor:
```
this.ngControl.valueAccessor = this;
```
By assigning it to `this`, the accessor will pickup all its required methods (`writeValue`, `registerOnChange`, etc..`). No need to change those!